lib/statsd.js

```diff
@@ -42,26 +43,55 @@ var Client = function (host, port, prefix, suffix, globalize, cacheDns, mock,
       maxBufferSize : maxBufferSize,
       bufferFlushInterval: bufferFlushInterval,
       telegraf    : telegraf,
-      sampleRate    : sampleRate
+      sampleRate  : sampleRate,
+      protocol    : protocol
     };
   }
 
+  var createSocket = function createSocket(args) {
+    var socket;
+    var errMessage;
+
+    if (args.protocol === 'tcp') {
+      try {
+        socket = net.connect(args.port, args.host);
+        socket.setKeepAlive(true);
+      } catch (e) {
+        errMessage = 'Could not establish connection to ' + args.host + ':' + args.port;
+        if (this.errorHandler) {
+          this.errorHandler(new Error(errMessage));
+        } else {
+          console.log(errMessage);
+        }
+      }
+    } else {
+      socket = dgram.createSocket('udp4');
+    }
+
+    return socket;
+  }.bind(this);
```

Comment:
Doing it this way means we have two ways to have functions in here.  Could you make this like the existing "_send" method?  Alternatively, if you feel strongly about it, it's not that I'm a big fan of "_send"- you could also make that method like this one.

Reply:
I added the `bind` to appease `jshint`, which is not good enough reason, I agree. I changed the `createSocket` function signature to accept instance argument instead (https://github.com/brightcove/hot-shots/pull/79/commits/dee23b3c6c80921182133127995ac74caf7d5bc4).